### ISSUE TITLE
Overlap arrays

### ISSUE THREAD

**Issue by phael-tavares** [prisma/prisma#11445]
## Problem

I've got an array field in a table and I want to filter the results of my select query using another array. If the record of the database has any of the elements of that "filtering" array, this record must be retrieved . Basically what an overlap function in SQL does.

## Suggested solution

Implement an overlap feature.

## Alternatives

Looked up on the internet and it seems to be the best approach for this filtering case.

## Additional context

Here is an example of overlaping in postgreSQL:

select
array[1,2,3] && array[3,7,9] as compare1,
array[1,2,3] && array[5,7,9] as compare2,
array['futebol','skate','cerveja'] && array['cerveja','cachaça','vinho'] as compare3,
array['futebol','skate','cerveja'] && array['cachaça','vinho'] as compare4;

compare1 = true
compare2 = false
compare 3 = true
compare 4 = false

**Comment by matthewmueller** [comment]
Interesting, I hadn't heard of this. Is that the same as https://andreigridnev.com/blog/2016-03-30-overlaps-operator-in-postgresql/?

**Comment by phael-tavares** [comment]
> Interesting, I hadn't heard of this. Is that the same as https://andreigridnev.com/blog/2016-03-30-overlaps-operator-in-postgresql/?

yeah I guess it's pretty much the same logic, but the "&&" operator is used for arrays while the "OVERLAPS" one is for dates :)
